cognite/neat/core/workflow/manager.py

```diff
@@ -127,6 +145,100 @@ def load_workflows_from_storage_v2(self, dir_path: str = None):
                             self.workflow_registry[wf_module_name].set_storage_path(
                                 "transformation_rules", self.rules_storage_path
                             )
+                            self.workflow_registry[wf_module_name].set_storage_path("data_store", self.data_store_path)
                 except Exception as e:
                     trace = traceback.format_exc()
                     logging.error(f"Error loading workflow {wf_module_name}: error: {e} trace : {trace}")
+
+    def create_workflow_instance(self, template_name: str, add_to_registry: bool = True) -> BaseWorkflow:
+        new_instance = self.workflow_registry[template_name].__class__(template_name, self.client)
+        new_instance.workflow_steps = self.workflow_registry[template_name].workflow_steps
+        new_instance.configs = self.workflow_registry[template_name].configs
+        new_instance.set_task_builder(self.task_builder)
+        new_instance.set_default_dataset_id(self.data_set_id)
+        new_instance.set_storage_path("transformation_rules", self.rules_storage_path)
+        new_instance.set_storage_path("data_store", self.data_store_path)
+        if add_to_registry:
+            self.ephemeral_instance_registry[new_instance.instance_id] = new_instance
+        live_workflow_intances.labels(itype="ephemeral").set(len(self.ephemeral_instance_registry))
+        return new_instance
+
+    def get_workflow_instance(self, instance_id: str) -> BaseWorkflow:
+        return self.ephemeral_instance_registry[instance_id]
+
+    def delete_workflow_instance(self, instance_id: str):
+        del self.ephemeral_instance_registry[instance_id]
+        live_workflow_intances.labels(itype="ephemeral").set(len(self.ephemeral_instance_registry))
+        return
+
+    def start_workflow_instance(
+        self, workflow_name: str, step_id: str = "", flow_msg: FlowMessage = None, sync: bool = None
+    ) -> WorkflowStartStatus:
```

Comment:
This is strange to me. I think `start_workflow_instance `should return a workflow which has a status field.

Reply:
Originally it was returning workflow but had to wrap it up into Status object as there are cases when workflow can't be started and it's good to be able to return explanation of why it can't be started. For instance existing instance already running or wait time expired .